pkg/syncproto/sync_proto.go

```diff
@@ -74,23 +73,25 @@ func SerializeUpdate(u api.Update) (kv SerializedUpdate, err error) {
 			"Bug: failed to serialize key that was generated by Syncer.")
 		return
 	}
-	kv.Value, err = model.SerializeValue(&u.KVPair)
-	if err != nil {
-		log.WithError(err).WithField("kv", u).Error(
-			"Bug: failed to serialize value, converting to nil value.")
-		var nilKV model.KVPair
-		nilKV.Key = u.Key
-		kv.Value, err = model.SerializeValue(&nilKV)
-		if err != nil {
-			log.WithError(err).WithField("kv", u.Key).Error(
-				"Bug: Failed to serialize nil value for key.")
-			return
-		}
-	}
 
 	kv.TTL = u.TTL
 	kv.Revision = u.Revision // This relies on the revision being a basic type.
 	kv.UpdateType = u.UpdateType
+
+	if u.Value == nil {
+		log.Debug("Value is nil, passing through as a deletion.")
+		return
+	}
+
+	value, err := model.SerializeValue(&u.KVPair)
+	if err != nil {
+		log.WithError(err).WithField("kv", u).Error(
```

Comment:
Same above too - it will be confusing to someone debugging that `"kv"` in the logging does not correspond to `kv` in the code.